langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiChatModel.java

```diff
@@ -152,7 +151,7 @@ public OpenAiChatModel(String baseUrl,
         this.strictTools = getOrDefault(strictTools, false);
         this.parallelToolCalls = parallelToolCalls;
         this.maxRetries = getOrDefault(maxRetries, 3);
-        this.tokenizer = getOrDefault(tokenizer, OpenAiTokenizer::new);
+        this.tokenizer = getOrDefault(tokenizer, new OpenAiTokenizer(this.modelName));
```

Comment:
`OpenAiTokenizer` should be created lazily

Reply:
Thank you for heads-upI will revert to previous code. I thought deprecated constructor should have been replaced